mesonbuild/modules/cmake.py

```diff
@@ -416,5 +542,91 @@ def subproject(self, state, args, kwargs):
+                raise mesonlib.MesonException('cmake.generate_export only supports dependencies wrapping a single library')
+
+            target = dep.libraries[0]
+            targets.append(target)
+            expected_targets += namespace + '::' + target.name + ' '
+
+            target_compile_options = ''
+            for compile_arg in dep.compile_args:
+                target_compile_options += compile_arg + ' '
+            compile_options.append(target_compile_options.strip())
+
+            target_link_options = ''
+            for link_arg in dep.link_args:
+                target_link_options += link_arg + ' '
+            link_options.append(target_link_options.strip())
+        expected_targets = expected_targets.strip()
+
+        name = kwargs['name']
+        if not name:
+            if len(deps) > 1:
+                raise mesonlib.MesonException('When passing multiple deps to cmake.generate_export, the name kwarg is required')
+            name = deps[0].libraries[0].name
+
+        includedir = coredata.get_option(mesonlib.OptionKey('includedir'))
+        include_dirs = f'"${{_IMPORT_PREFIX}}/{includedir}" '
+        for subdir in kwargs['subdirs']:
+            include_dirs += f'"${{_IMPORT_PREFIX}}/{includedir}/{subdir}" '
+        include_dirs = include_dirs.strip()
+
+        targets_file = TARGETS_INIT.replace('@expected_targets@', expected_targets)
+        targets_file = targets_file.replace('@prefix@', coredata.get_option(mesonlib.OptionKey('prefix')))
+
+        buildtype: str = coredata.get_option(mesonlib.OptionKey('buildtype')).upper()
+        targets_config_file = TARGETS_CONFIG_INIT.replace('@buildtype@', buildtype)
+
+        for i, target in enumerate(targets):
+            targets_file += TARGETS_TARGET.replace('@namespace@', namespace)
+            targets_file = targets_file.replace('@target_name@', target.name)
+
+            lib_type = 'SHARED' if isinstance(target, build.SharedLibrary) else 'STATIC'
+            targets_file = targets_file.replace('@lib_type@', lib_type)
+
+            targets_file = targets_file.replace('@compile_options@', compile_options[i])
+            targets_file = targets_file.replace('@include_dirs@', include_dirs)
+            #targets_file = targets_file.replace('@libraries@', '')
+            targets_file = targets_file.replace('@link_options@', link_options[i])
+
+            targets_config_file += TARGETS_CONFIG_TARGET.replace('@namespace@', namespace)
+            targets_config_file = targets_config_file.replace('@target_name@', target.name)
+            targets_config_file = targets_config_file.replace('@buildtype@', buildtype)
+            targets_config_file = targets_config_file.replace('@libdir@', coredata.get_option(mesonlib.OptionKey('libdir')))
+            targets_config_file = targets_config_file.replace('@lib_file_name@', target.filename)
```

Comment:
When building a shared library, this will generate an imported location pointing to the versioned library (e.g. `libname.so.1.2`). This matches CMake's behaviour, but it is wrong as users should link against the unversioned symlink. Is there a way to get the unversioned file name from the library?

Reply:
Yeah, you can generate `-lfoo` link arguments just like pkg-config does. :)

Reply:
I can't, I need the actual file name of the library because CMake needs the full path to the compiled library that it needs to link against to be set in the `IMPORTED_PATH_<config>` property
https://github.com/mesonbuild/meson/blob/8c5e76eee00bb6dddc5429a49ce8ca6aefd29563/mesonbuild/modules/cmake.py#L201